File: android-http/java/com/hmkcode/android/http/MainActivity.java
  package com.hmkcode.android.http;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

  public class MainActivity extends AppCompatActivity {

    TextView tvIsConnected;
    TextView tvResult;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tvIsConnected = (TextView) findViewById(R.id.tvIsConnected);
        tvResult = (TextView) findViewById(R.id.tvResult);
        if(checkNetworkConnection())
            // perform HTTP GET request
            new HTTPAsyncTask().execute("http://hmkcode.com/examples/index.php");
    }


    // check network connection
    public boolean checkNetworkConnection() {
        ConnectivityManager connMgr = (ConnectivityManager)
                getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
        boolean isConnected = false;
        if (networkInfo != null && (isConnected = networkInfo.isConnected())) {
            // show "Connected" & type of network "WIFI or MOBILE"
            tvIsConnected.setText("Connected "+networkInfo.getTypeName());
            // change background color to red
            tvIsConnected.setBackgroundColor(0xFF7CCC26);


        } else {
            // show "Not Connected"
            tvIsConnected.setText("Not Connected");
            // change background color to green
            tvIsConnected.setBackgroundColor(0xFFFF0000);
        }

        return isConnected;
    }

  private class HTTPAsyncTask extends AsyncTask<String, Void, String> {
      @Override
      protected String doInBackground(String... urls) {

          // params comes from the execute() call: params[0] is the url.
          try {
              return HttpGet(urls[0]);
          } catch (IOException e) {
              return "Unable to retrieve web page. URL may be invalid.";
          }
      }
      // onPostExecute displays the results of the AsyncTask.
      @Override
      protected void onPostExecute(String result) {
          tvResult.setText(result);
      }
  }

  private String HttpGet(String myUrl) throws IOException {
      InputStream inputStream = null;
      String result = "";

          URL url = new URL(myUrl);

          // create HttpURLConnection
          HttpURLConnection conn = (HttpURLConnection) url.openConnection();

          // make GET request to the given URL
          conn.connect();

          // receive response as inputStream
          inputStream = conn.getInputStream();

          // convert inputstream to string
          if(inputStream != null)
              result = convertInputStreamToString(inputStream);
          else
              result = "Did not work!";

      return result;
  }

      private static String convertInputStreamToString(InputStream inputStream) throws IOException{
          BufferedReader bufferedReader = new BufferedReader( new InputStreamReader(inputStream));
          String line = "";
          String result = "";
          while((line = bufferedReader.readLine()) != null)
              result += line;

          inputStream.close();
          return result;

      }


  }

File: android-http/app/src/main/java/com/hmkcode/http/MainActivity.kt
package com.hmkcode.http

import android.content.Context
import android.net.ConnectivityManager
import android.net.Network
import android.net.NetworkInfo
import android.os.AsyncTask
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class MainActivity : AppCompatActivity() {

    lateinit var tvIsConnected: TextView;
    lateinit var tvResult: TextView;


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvIsConnected = findViewById<TextView>(R.id.tvIsConnected)
        tvResult = findViewById<TextView>(R.id.tvResult)

        if(checkNetworkConnection())
            HTTPAsyncTask().execute("http://hmkcode-api.appspot.com/rest/api/hello/Android")
    }


    private fun checkNetworkConnection(): Boolean {
        val cm:ConnectivityManager =
            getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val networkInfo:NetworkInfo? = cm.activeNetworkInfo
        val isConnected: Boolean = if(networkInfo != null) networkInfo.isConnected() else false

        if(isConnected){
            tvIsConnected.setText("Connected "+networkInfo?.typeName)
            tvIsConnected.setBackgroundColor(0xFF7CCC26.toInt())

        }else{
            tvIsConnected.setText("Not Connected!")
            tvIsConnected.setBackgroundColor(0xFFFF0000.toInt())
        }
        return isConnected;
    }

    inner class HTTPAsyncTask : AsyncTask<String, Void, String>() {
        override fun doInBackground(vararg urls: String?): String {
            return HttpGet(urls[0])
        }
        override fun onPostExecute(result: String?) {
            tvResult.setText(result)
        }
    }

    private fun HttpGet(myURL: String?): String {

        val inputStream:InputStream
        val result:String

        // create URL
        val url:URL = URL(myURL)

        // create HttpURLConnection
        val conn:HttpURLConnection = url.openConnection() as HttpURLConnection

        // make GET request to the given URL
        conn.connect()

        // receive response as inputStream
        inputStream = conn.inputStream

        // convert inputstream to string
        if(inputStream != null)
            result = convertInputStreamToString(inputStream)
        else
            result = "Did not work!"
        
        return result
    }

    private fun convertInputStreamToString(inputStream: InputStream): String {
        val bufferedReader:BufferedReader? = BufferedReader(InputStreamReader(inputStream))
        var line:String? = bufferedReader?.readLine()
        var result:String = ""

        while (line != null) {
            result += line
            line = bufferedReader?.readLine()
        }

        inputStream.close()
        return result
    }

}
